docs/topics/kotlin-daemon.md

```diff
@@ -0,0 +1,106 @@
+[//]: # (title: Kotlin daemon)
+
+The Kotlin daemon is a background process that build systems can use to improve Kotlin compilation performance.
+The Kotlin daemon speeds up build times by keeping the compiler and its environment ready to compile. This approach 
+avoids the overhead of starting a new Java Virtual Machine (JVM) instance and reinitializing the compiler for every 
+compilation, leading to reduced build times for incremental builds or frequent small changes.
+
+Some build systems, like [Gradle](https://docs.gradle.org/current/userguide/gradle_daemon.html) and [Maven](https://maven.apache.org/tools/mvnd.html),
+have their own daemons that help reduce startup costs. However, they miss the added benefit that the Kotlin daemon
+completely isolates the build system process from the compiler. This separation is useful in dynamic environments where
+system settings might change at runtime.
+
+Although there's no official user-facing interface for the daemon, it can be used by build systems or through the [build tools API](build-tools-api.md).
```

Comment:
I'm only not sure about this wording.
It's expected that build system plugins may expose daemon functionality, but they also are expected to work through the Build Tools API